packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx

```diff
@@ -0,0 +1,477 @@
+          containerClassName={`input ${inputError ? 'failure' : ''}`}
+          value={inputValue}
+          onInput={(e) => onInputValueChange(e.target?.value?.trim())}
+        />
+      }{' '}
+      # of all requirements
+    </span>
+  );
+
+  return (
+    <CWRadioButton
+      label={Label}
+      value="N"
+      name="requirementsToFulfill"
+      hookToForm
+    />
+  );
+};
+
+const MAX_REQUIREMENTS = 10;
+
+const GroupForm = ({
+  formType,
+  onSubmit,
+  initialValues = {},
+}: GroupFormProps) => {
+  const { data: topics } = useFetchTopicsQuery({
+    chainId: app.activeChainId(),
+  });
+  const sortedTopics = (topics || []).sort((a, b) => a?.name?.localeCompare(b));
+  const [cwRequiremnetsLabelInputField, setCWRequiremnetsLabelInputField] =
+    useState<{ value: string; error: string }>({ value: '1', error: '' });
+  const [requirementSubForms, setRequirementSubForms] = useState<
+    {
+      defaultValues?: RequirementSubTypeWithLabel;
+      values: RequirementSubType;
+      errors?: RequirementSubType;
+    }[]
+  >([
+    {
+      values: {
+        requirementAmount: '',
+        requirementChain: '',
+        requirementCondition: '',
+        requirementContractAddress: '',
+        requirementType: '',
+      },
+      errors: {},
+    },
+  ]);
+
+  useEffect(() => {
+    if (initialValues.requirements) {
+      setRequirementSubForms(
+        initialValues.requirements.map((x) => ({
+          defaultValues: x,
+          values: {
+            requirementAmount: x?.requirementAmount || '',
+            requirementChain: x?.requirementChain?.value || '',
+            requirementCondition: x?.requirementCondition?.value || '',
+            requirementContractAddress: x?.requirementContractAddress || '',
+            requirementType: x?.requirementType?.value || '',
+          },
+          errors: {},
+        }))
+      );
+    }
+
+    if (
+      initialValues.requirementsToFulfill &&
+      initialValues.requirementsToFulfill !== 'ALL'
+    ) {
+      setCWRequiremnetsLabelInputField({
+        ...cwRequiremnetsLabelInputField,
+        value: `${initialValues.requirementsToFulfill}`,
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const removeRequirementByIndex = (index: number) => {
+    setRequirementSubForms(requirementSubForms.splice(index, 1));
+  };
+
+  const addRequirementSubForm = () => {
+    setRequirementSubForms([
+      ...requirementSubForms,
+      {
+        values: {
+          requirementAmount: '',
+          requirementChain: '',
+          requirementCondition: '',
+          requirementContractAddress: '',
+          requirementType: '',
+        },
+        errors: {},
+      },
+    ]);
+  };
+
+  const validateChangedValue = (val, index) => {
+    const allRequirements = [...requirementSubForms];
+
+    // HACK ALERT: this type of validation change should be done internally by zod, by we are doing this manually using javascript
```

Comment:
> I am ok with this "hack". Don't even know if this is a hack ;)

In terms of our "better form" design, I would consider it a hack, the purpose of the newer forms design was to make life easy but these kind of ad-hoc validations will do the opposite. To reduce friction, I believe it can go in for now.

> The only thing I would improve here is that quite a lot of the code is being repeated 3 times. Maybe you could create some util function in separate file or custom hook that will do the heavy lifting, so this component won't be so polluted with custom code?

done